markdown_flow/providers/openai.py

```
@@ -0,0 +1,369 @@
+            response = self.client.chat.completions.create(
+                model=actual_model,
+                messages=formatted_messages,
+                temperature=actual_temperature,
+            )
+
+            # Calculate processing time
+            processing_time_ms = int((time.time() - start_time) * 1000)
+
+            # Extract content
+            if not response.choices or len(response.choices) == 0:
+                raise Exception("API response error: no choices returned")
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)